pkg/capture/capture_manager.go

```diff
@@ -360,11 +360,16 @@ func (cm *Manager) update(ctx context.Context, ifaces config.Ifaces, enable, dis
 			logger.Info("initializing capture / running packet processing")
 
 			newCap := newCapture(iface, ifaces[iface]).SetSourceInitFn(cm.sourceInitFn)
-			if err := newCap.run(runCtx); err != nil {
+			if err := newCap.run(); err != nil {
 				logger.Errorf("failed to start capture: %s", err)
 				return
 			}
 
+			// Start up processing and error handling / logging in the
+			// background
+			go cm.logErrors(runCtx, iface,
```

Comment:
I had to move tracking / logging to the `CaptureManager` (instead of handling the goroutine startup from the capture itself) because only this way can we manipulate the map of captures when processing has terminated prematurely (see below in `logErrors(...)` implementation).